app/scripts/components/common/map/controls/hooks/use-draw-control.ts

```diff
@@ -0,0 +1,83 @@
+import { useState, useEffect, useRef } from 'react';
+import MapboxDraw from '@mapbox/mapbox-gl-draw';
+import '@mapbox/mapbox-gl-draw/dist/mapbox-gl-draw.css';
+import { Feature } from 'geojson';
+
+interface DrawAOIProps {
+  mapboxMap: mapboxgl.Map | null;
+  isDrawing: boolean;
+  styles: Array<object>;
+}
+
+interface DrawControlState {
+  drawing: Feature | Feature[] | null;
+  isValid: boolean;
+}
+
+/**
+ * Custom hook to manage drawing control on a Mapbox GL map.
+ *
+ * @param {mapboxgl.Map | null} props.mapboxMap - The Mapbox GL map instance.
+ * @param {boolean} props.isDrawing - Flag indicating whether drawing mode is active.
+ * @returns {[Feature | Feature[] | null, boolean]} - An array containing the current drawing and its validity.
+ *
+ * @example
+ * const [drawing, isValid] = useDrawControl({ mapboxMap, isDrawing });
+ */
+export const useDrawControl = ({
+  mapboxMap,
+  isDrawing,
+  styles
+}: DrawAOIProps): [Feature | Feature[] | null, boolean] => {
+  const [{ drawing, isValid }, setDrawing] = useState<DrawControlState>({
+    drawing: null,
+    isValid: false
```

Comment:
Real nitpick: It could be because I am so used to the term 'validation of' geometry, but this variable name throws me slightly off. As far as I understand, this flag is an explicit variable to save whether the drawing feature is null or not, which I don't think my brain clicks to 'valid' status.

Reply:
I am using this flag to enable the 'confirm area' button after a polygon has been drawn. Confirming is not possible while still drawing the shape.

Reply:
Yah, I found that it is used there later. (I edited my comment later in case you saw the outdated one 😓 ) my comment is more about whether the variable can be named differently.